agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -62,6 +88,16 @@ class AtumContext private[agent] (
     )
   }
 
+  def createAndSaveCheckpointOnProvidedData(
+    checkpointName: String, author: String, measurements: Seq[Measurement]
```

Comment:
Agreed to potentially change this once we'll be refactoring the measurement builder & Measure class - all this will be changed to some degree (some changes will be more brutal) in https://github.com/AbsaOSS/atum-service/issues/98